pilot/pkg/networking/core/v1alpha3/listener_inbound.go

```diff
@@ -412,7 +412,8 @@ func (lb *ListenerBuilder) getFilterChainsByServicePort(chainsByPort map[uint32]
 		}
 		actualWildcards, _ := getWildcardsAndLocalHost(lb.node.GetIPMode())
 		if enableSidecarServiceInboundListenerMerge && sidecarScope.HasIngressListener() &&
-			ingressPortListSet.Contains(int(port.Port)) {
+			// ingress listener port means the target port, may not equal to service port
+			ingressPortListSet.Contains(int(port.TargetPort)) {
```

Comment:
I am a bit confused, Port and TargetPort are always the same, right? So is this just making it more clear, or fixing a big?

Reply:
This is fixing a bug, as you can see in L407~412. The port and target port are for service concept, can be different

```
		port := ServiceInstancePort{
			Name:       i.ServicePort.Name,
			Port:       uint32(i.ServicePort.Port),
			TargetPort: i.Endpoint.EndpointPort,
			Protocol:   i.ServicePort.Protocol,
		}```